lib/rubocop/cop/rspec/rails/persistence_called_outside_example.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Rails
+        # Checks for persistence calls outside example blocks.
+        #
+        # Prevents persistence calls outside examples which usually run
+        # with some sort of cleanup hook. Saving outside of example causes
+        # records to leak into other tests.
+        #
+        # @ example
+        #   # bad - records created outside example group
+        #   describe User do
+        #     User.create!
+        #   end
+        #
+        #   describe User do
+        #     user = User.new
+        #     user.save!
+        #   end
+        #
+        #   # good - records created inside an example group
+        #   describe User do
+        #     it do
+        #       User.create!
+        #     end
+        #   end
+        #
+        #   describe User do
+        #     it do
+        #       user = User.new
+        #       user.save!
+        #     end
+        #   end
+        class PersistenceCalledOutsideExample < Base
+          MSG = 'Persistence called outside of example.'
+
+          def on_send(node)
+            return if inside_example_scope?(node) ||
+              inside_method_definition?(node) ||
+              inside_proc_or_lambda?(node) ||
```

Comment:
Maybe extract those three into `lazily_executed?` (pretty sure a better for such a method name exists) to appease CodeClimate?